### PR TITLE
NASNet-A Implementation/Checkpoints

### PR DESCRIPTION
**Pull request by sguada**
This addition contains the code for the NASNet-A model from the paper Learning Transferable Architectures for Scalable Image Recognition by Zoph et al. In nasnet.py there are three different configurations of NASNet-A that are implemented.

### CODE REVIEW

**Comment by sguada** [comment]
I sign it

File: research/slim/README.md [hunk 1]
```diff
@@ -127,7 +126,7 @@ from integer labels to class names.
 
 You can use the same script to create the mnist and cifar10 datasets.
 However, for ImageNet, you have to follow the instructions
-[here](https://github.com/tensorflow/models/blob/master/research/inception/README.md#getting-started).
```
Comment (derekjchow) [inline research/slim/README.md]: Please revert.

File: research/slim/datasets/cifar10.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 """Provides data for the Cifar10 dataset.
 
 The dataset scripts used to create the dataset can be found at:
-tensorflow/models/research/slim/datasets/download_and_convert_cifar10.py
```
Comment (derekjchow) [inline research/slim/datasets/cifar10.py]: Please revert.

File: research/slim/datasets/download_and_convert_imagenet.sh [hunk 3]
```diff
@@ -58,7 +58,7 @@ DATA_DIR="${1%/}"
 SCRATCH_DIR="${DATA_DIR}/raw-data/"
 mkdir -p "${DATA_DIR}"
 mkdir -p "${SCRATCH_DIR}"
-WORK_DIR="$0.runfiles/third_party/tensorflow_models/research/slim"
```
Comment (derekjchow) [inline research/slim/datasets/download_and_convert_imagenet.sh]: Please revert.

File: research/slim/datasets/flowers.py [hunk 4]
```diff
@@ -15,7 +15,7 @@
 """Provides data for the flowers dataset.
 
 The dataset scripts used to create the dataset can be found at:
-tensorflow/models/research/slim/datasets/download_and_convert_flowers.py
```
Comment (derekjchow) [inline research/slim/datasets/flowers.py]: Revert

File: research/slim/datasets/mnist.py [hunk 5]
```diff
@@ -15,7 +15,7 @@
 """Provides data for the MNIST dataset.
 
 The dataset scripts used to create the dataset can be found at:
-tensorflow/models/research/slim/datasets/download_and_convert_mnist.py
```
Comment (derekjchow) [inline research/slim/datasets/mnist.py]: Ditto (sorry for the repetition!).

Maybe do a grep for models/slim and find the remaining instances.

File: research/slim/nets/cyclegan.py [hunk 6]
```diff
@@ -0,0 +1,269 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
```
Comment (derekjchow) [inline research/slim/nets/cyclegan.py]: We have more goodies than just NASnet =D. Maybe add these to your commit message.

File: research/slim/nets/mobilenet_v1.py [hunk 7]
```diff
@@ -154,7 +154,7 @@ def mobilenet_v1_base(inputs,
     inputs: a tensor of shape [batch_size, height, width, channels].
     final_endpoint: specifies the endpoint to construct the network up to. It
       can be one of ['Conv2d_0', 'Conv2d_1_pointwise', 'Conv2d_2_pointwise',
-      'Conv2d_3_pointwise', 'Conv2d_4_pointwise', 'Conv2d_5_pointwise',
+      'Conv2d_3_pointwise', 'Conv2d_4_pointwise', 'Conv2d_5'_pointwise,
```
Comment (derekjchow) [inline research/slim/nets/mobilenet_v1.py]: Please revert (non research/ related).

File: research/slim/nets/pix2pix.py [hunk 8]
```diff
@@ -0,0 +1,292 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
```
Comment (derekjchow) [inline research/slim/nets/pix2pix.py]: More goodies.

File: research/slim/slim_walkthrough.ipynb [hunk 9]
```diff
@@ -253,7 +253,7 @@
     "    # Add the loss function to the graph.\n",
     "    loss = tf.losses.mean_squared_error(labels=targets, predictions=predictions)\n",
     "    \n",
-    "    # The total loss is the user's loss plus any regularization losses.\n",
+    "    # The total loss is the uers's loss plus any regularization losses.\n",
```
Comment (derekjchow) [inline research/slim/slim_walkthrough.ipynb]: Typo here.

**Comment by jhseu** [comment]
(It looks like this accidentally deleted everything.)

**Comment by ebrevdo** [comment]
There seems to be something very wrong with this PR; it deletes a *lot* of files (900k lines of code).  Perhaps you need to rebase on the most recent snapshot of models and ensure you properly patch in just the necessary file changes?

**Comment by sguada** [comment]
Sorry

**Pull request closed without merging**